File: 0001-two-sum/TwoSum.swift
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var m = [Int: Int]()
        for (i, v) in nums.enumerated() {
            let complement = target - v
            if let j = m[complement] {
                return [j, i]
            }
            m[v] = i
        }
        return []
    }

File: 0000-0099/0001-two-sum/two-sum.go
package solution

func twoSum(nums []int, target int) []int {
	m := make(map[int]int, len(nums))
	for i, v := range nums {
		complement := target - v
		if _, ok := m[complement]; ok {
			return []int{m[complement], i}
		}
		m[v] = i
	}
	return nil
}
